src/gtk/control.cpp

```diff
@@ -99,6 +103,9 @@ void wxControl::PostCreation(const wxSize& size)
 #ifdef __WXGTK3__
     if (HasFlag(wxNO_BORDER))
         GTKApplyCssStyle("*{ border:none; border-radius:0; padding:0 }");
+
+    // Cache the preferred size.
+    m_cachedPreferredSize = GTKGetPreferredSize(m_widget);
```

Comment:
Do we really need to do this? It looks like `DoGetBestSize()`, and hence `GTKGetPreferredSize()`, will be called for any visible window anyhow very quickly, and if it's not visible, it's useless to call it anyhow...